### PR TITLE
Missing error codes

### PR DESCRIPTION
**Pull request by biafra23**
Handles #1120 

Changes proposed in this pull request:

- Parse body for error code for 422 response code
- Treat unmatched 3xx, 4xx and 5xx the same but preserve http response code as error code
- Tracking of errors
- Unit tests added

@gnosis/mobile-devs

### CODE REVIEW

File: data/src/main/java/io/gnosis/data/backend/TransactionSerivceApi.kt [hunk 1]
```diff
@@ -8,7 +8,7 @@ import retrofit2.http.Path
 @Deprecated("use client gw")
 interface TransactionServiceApi {
 
-    @GET("v1/safes/{address}")
+    @GET("v1/safes/{address}/")
```
Comment (biafra23) [inline data/src/main/java/io/gnosis/data/backend/TransactionSerivceApi.kt:11]: Prevent redirect. We should check all endpoints for redirects.

File: app/src/main/java/io/gnosis/safe/ErrorHandler.kt [hunk 2]
```diff
@@ -77,39 +105,73 @@ enum class Error(
...
     }
+
+@JsonClass(generateAdapter = true)
+data class ServerError(
+    @Json(name = "code") val code: Int,
+    @Json(name = "message") val message: String?,
+    @Json(name = "arguments") val arguments: List<String>?
+)
+
+private val serverErrorAdapter = dataMoshi.adapter(ServerError::class.java)
```
Comment (elgatovital) [inline app/src/main/java/io/gnosis/safe/ErrorHandler.kt:177]: why do we need this adapter here?

File: app/src/main/java/io/gnosis/safe/ErrorHandler.kt [hunk 3]
```diff
@@ -77,39 +105,73 @@ enum class Error(
...
+                        val errorBodyString = it.response()?.errorBody()?.string()
+                        val serverError = errorBodyString?.let {
+                            if (errorBodyString.isNotEmpty()) {
+                                serverErrorAdapter.fromJson(errorBodyString)
+                            } else {
+                                null
+                            }
+                        }
+
+                        if (Tracker.isInitialized()) {
```
Comment (elgatovital) [inline app/src/main/java/io/gnosis/safe/ErrorHandler.kt:139]: as discussed let's have tracking logic at places where mapping is called. BaseFragment and BaseActivity already have tracker property.

Comment (biafra23) [reply app/src/main/java/io/gnosis/safe/ErrorHandler.kt]: Done in: 2a2648867868a5b1437085a01e4898229f1b77cf

**Review by elgatovital — APPROVED** [review]

**Pull request merged**